// @storyblok/react/dist/ssr.mjs
import { storyblokInit as r } from "./core/init.mjs";
import { useStoryblokApi as l, useStoryblokApi as p } from "./core/use-storyblok-api.mjs";
import { default as a } from "./core/storyblok-component.mjs";
import { useStoryblokState as k } from "./core/use-storyblok-state.mjs";
import { BlockTypes as S, MarkTypes as m, TextTypes as n, apiPlugin as i, loadStoryblokBridge as x, registerStoryblokBridge as f, renderRichText as u, richTextResolver as d, storyblokEditable as g, useStoryblokBridge as T } from "@storyblok/js";
import { useStoryblokServerRichText as C } from "./server/richtext.mjs";
import { default as h } from "./server/server-component.mjs";
import { default as R } from "./server/server-story.mjs";
import { default as E } from "./server/server-storyblok-richtext-component.mjs";
import { getComponent as I, getCustomFallbackComponent as M, getEnableFallbackComponent as P, setComponents as j } from "./core/state.mjs";
export {
  S as BlockTypes,
  m as MarkTypes,
  a as StoryblokComponent,
  h as StoryblokServerComponent,
  E as StoryblokServerRichText,
  R as StoryblokServerStory,
  n as TextTypes,
  i as apiPlugin,
  I as getComponent,
  M as getCustomFallbackComponent,
  P as getEnableFallbackComponent,
  l as getStoryblokApi,
  x as loadStoryblokBridge,
  f as registerStoryblokBridge,
  u as renderRichText,
  d as richTextResolver,
  j as setComponents,
  g as storyblokEditable,
  r as storyblokInit,
  p as useStoryblokApi,
  T as useStoryblokBridge,
  C as useStoryblokServerRichText,
  k as useStoryblokState
};
